text/0000-rename-distributed-ctx.md

```diff
@@ -0,0 +1,59 @@
+# Rename distributed context into tags and add support for Baggage.
+
+**Status:** `proposed`
+
+Rename Distributed-context to a less confusing term and restrict its use for observability.
+ 
+ 
+## Motivation
+
+Context Propagation is generalized to carry observability and non-observability contexts. This
+includes SpanContext, Distributed-context, Authentication-context, etc. Distributed-context
+carries key-value pairs for observability. The name Distributed-context is
+very confusing. Hence it should be changed to TagMap (or other better suggestion). 
+That is the first motivation.
+ 
+The second motivation is the restrict use of Distributed-context to observability. As it is
+written currently it does not specifically call out for using it only for non-observability purpose.
+However, it can be used to support 'Baggage' functionality for non-observability purpose as it is
+OpenTracing.
+There are few reasons to have different mechanism to address observability and
+non-observability use cases.
+1. For observability, key-value pairs SHOULD be write-only from application perspective. SDKs can read them
+   for example to apply them to measurements at the time of recording. For non-observability, 
+   key-value pairs SHOULD be writable/readable for application.
+2. For non-observability, it may require that the delivery is guaranteed and pairs are not dropped.
+3. Semantically two are different and mechanically it could be different as well.
+
+## Explanation
+
+We do following to
+- Rename distributed-context to TagMap.
+  - It is strictly used for observability.
+  - It is a write-only object from application and framework-integrations perspective.
+  - It is readable by SDKs.
+
+- Add Baggage support. Equivalent of what is there in [OpenTracing](https://opentracing.io/docs/overview/tags-logs-baggage/#baggage-items).
```

Comment:
It is unclear when to put an item in Baggage versus TagMap.
Also, if something started as an item in TagMap, later there are needs to also put it Baggage, do we move the item or we duplicate the item?

A meta question - do we believe that OpenTelemetry should provide such general purpose `Baggage` so people can use it for things such as "authentication context"?

Reply:
> It is unclear when to put an item in Baggage versus TagMap.

As I read this OTEP, that is quite clear: An application puts a value in the TagMap if it wants to have spans, metrics, etc tagged with that value. The baggage on the other hand is used when the value should not be attached to the telemetry data but if the application itself (or another service invoked by it) wants to read the value back, for application-specific purposes.

Reply:
> Also, if something started as an item in TagMap, later there are needs to also put it Baggage, do we move the item or we duplicate the item?

No. Two are independent. Yes, there is duplication as mentioned in Trade-offs section but it is not worth to optimize it.

> 
> A meta question - do we believe that OpenTelemetry should provide such general purpose `Baggage` so people can use it for things such as "authentication context"?

I agree that from observability perspective, 'Baggage' doesn't belong in OpenTelemetry. However, from POV of opentracing compatibility we need to provide some solution. It could very well be in another repo/contrib package.

Reply:
> No. Two are independent. Yes, there is duplication as mentioned in Trade-offs section but it is not worth to optimize it.

I see, now it is clear to me. Thanks.

> I agree that from observability perspective, 'Baggage' doesn't belong in OpenTelemetry. However, from POV of opentracing compatibility we need to provide some solution. It could very well be in another repo/contrib package.

Makes sense. If this for compatibility, do we encourage more people to adopt it, or we suggest people to move away from it?

Reply:
Baggage is just context propagation. Context propagation is not a part of OpenTelemetry, it's the layer on top of which OpenTelemetry is built.

Reply:
I would vote for not including a non-telemetry related concept directly into opentelemetry. If Baggage is needed for opentracing having it be in contrib makes sense.

Will SaaS vendors need to use this as well? authentication or identification seems like a common concept.